test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js

```diff
@@ -115,33 +115,35 @@ define([
             expect(Jupyter.notebook.delete_cell).toHaveBeenCalled();
         });
 
-        it('responds to a delete-cell bus message', (done) => {
+        it('responds to a delete-cell bus message', () => {
             const runtime = Runtime.make();
             const cell = Mocks.buildMockCell('code');
-            Jupyter.notebook = Mocks.buildMockNotebook({
-                deleteCallback: () => {
-                    expect(Jupyter.notebook.delete_cell).toHaveBeenCalled();
-                    done();
-                },
-            });
-            spyOn(Jupyter.notebook, 'delete_cell').and.callThrough();
-            BulkImportCell.make({
-                cell,
-                importData: fakeInputs,
-                specs: fakeSpecs,
-                initialize: true,
-            });
-            runtime.bus().send(
-                {},
-                {
-                    channel: {
-                        cell: cell.metadata.kbase.attributes.id,
-                    },
-                    key: {
-                        type: 'delete-cell',
+            return new Promise((resolve) => {
```

Comment:
Rewrote this to work around some bus timing problems. Dealing with Promises seems to be more stable.

Reply:
Promising tests are definitely the way to go! ;)  I wish we had known that earlier.